code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java

```diff
@@ -225,6 +236,35 @@ protected void modifyDisk(long hostId, Instance instance, boolean add) {
 
     }
 
+    @Inject
+    IdFormatter idFormatter;
+
+    public void addReleaseSchedulerResource(String action, Long instanceId, boolean isVM, Long hostId, Long envId) {
```

Comment:
Can you check if we hold the lock in the caller stack? Or if the lock is held in your scheduler microservice, it is fine too